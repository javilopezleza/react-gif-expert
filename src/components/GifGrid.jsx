import PropTypes from "prop-types";
import { useState, useEffect } from "react";
// import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {
                isLoading  && (<h2>Cargando....</h2>)
            }

            <div className="card-grid">
                {
                    images.map((img) => (
                        <GifItem
                            key={img.id}
                            { ...img }
                        />
                    ))
                }
            </div>

        </>
    )
}

GifGrid.PropTypes = {
    category: PropTypes.string.isRequired,
}