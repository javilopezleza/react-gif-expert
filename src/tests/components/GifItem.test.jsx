import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";


describe('test in <GifItem />', () => {

    const title = 'Soy un titulo';
    const url = 'https://url-de-prueba.com/';

    test('Should do match with the snapshot', () => {

        const { container } = render(<GifItem
            title={title}
            url={url}
        />)

        expect(container).toMatchSnapshot();

    });

    test('should show the image with the appropiate URL and ALT ', () => {

        render(<GifItem title={title} url={url} />);
        // expect(screen.getByRole('img').src ).toBe( url );

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);

    });

    test('should show the title at the component', () => { 

        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();


     })

});