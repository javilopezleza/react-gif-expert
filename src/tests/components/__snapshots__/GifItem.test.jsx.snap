// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <GifItem /> Should do match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Soy un titulo"
      src="https://url-de-prueba.com/"
    />
    <p>
      Soy un titulo
    </p>
  </div>
</div>
`;
